call /system/etc/(post/pre)_sleep.sh before and after sleep, allow module probing

Kethen opened the pull request:
This will allow prebaked post s3 sleep workarounds

Kethen committed:
call /system/etc/(post/pre)_sleep.sh before and after sleep, allow module probing

hmtheboy154 commented:
Better putting them in /vendor or somewhere else

Kethen committed:
run suspend hal as root so that sleep hooks can do anything

Kethen commented:
> Better putting them in /vendor or somewhere else

Why? Suspend hal is a /system hal, it wouldn't make too much sense for it to fetch items from /vendor imo

hmtheboy154 commented:
> > Better putting them in /vendor or somewhere else
> 
> Why? Suspend hal is a /system hal, it wouldn't make too much sense for it to fetch items from /vendor imo

ok then check both on /vendor and /system. Because I wanna make it more treble

Kethen committed:
run /vendor/etc/(pre|post)_sleep.sh as well

because https://github.com/BlissRoms-x86/platform_system_hardware_interfaces/pull/7#issuecomment-1930082563 aka Huy Minh <39849246+hmtheboy154@users.noreply.github.com> said so
